assets/chat/js/autocomplete.js

```diff
@@ -241,8 +241,10 @@ class ChatAutoComplete {
     add(str, isemote = false) {
         const id = getBucketId(str);
         const bucket = this.buckets.get(id) || this.buckets.set(id, new Map()).get(id);
-        const data = Object.assign(bucket.get(str) || {}, {
+        const candidate = bucket.get(str);
+        const data = Object.assign(candidate || {}, {
```

Comment:
If the weight isn't updated through the `add` function anymore.
Then there is no need to keep around the updating of the `data` object.

Reply:
Indeed, latest commit checks for collision in the bucket before updating.